package/MDAnalysis/lib/qcprot.pyx

```diff
@@ -186,15 +192,15 @@ def InnerProduct(np.ndarray[np.float64_t,ndim=1] A,
 
     if (weight is not None):
         for i in range(N):
-            x1 = weight[i] * coords1[0,i]
-            y1 = weight[i] * coords1[1,i]
-            z1 = weight[i] * coords1[2,i]
+            x1 = weight[i] * coords1[i,0]
```

Comment:
Yeah we have c-style. And yes the current speed improvement is already nice. I'm interested if there is still some more head room. Memory layout and retrieving of memory that is easier for a CPU can make a lot of difference. Just have a look for cache friendly matrix multiplication algorithms (they take this to the very extreme)